herbert_core/classes/MPIFramework/@ClusterMPI/ClusterMPI.m

```diff
@@ -172,9 +172,20 @@ function check_availability(obj)
             rootpath = fileparts(which('herbert_init'));
             external_dll_dir = fullfile(rootpath, 'DLL','external');
             if ispc()
-                % only one version of mpiexec is used now. May change in the
-                % future.
-                mpi_exec = fullfile(external_dll_dir, 'mpiexec.exe');
+                [rs, rv] = system('where mpiexec');
```

Comment:
I tested it with [MSMPI v8](https://www.microsoft.com/en-us/download/confirmation.aspx?id=55494), [MSMPI v9](https://www.microsoft.com/en-us/download/details.aspx?id=56727) and [MSMPI v10](https://www.microsoft.com/en-us/download/details.aspx?id=100593) and it works on all 3 with the Jenkins compiled `mex` files - I just download the release bundle and use it as is; I don't recompile the `mex` files.

I think that the `cpp_communicator` only uses MPI2.2 (or earlier) features and that standard came out in 2009 and is supported by all three versions of MS MPI we're talking about here. It seems there are no ABI changes between the versions either so they all seem to work ok.

The issue with the Horace-bundled `mpiexec.exe` is due to the firewall (at least for me; I think). Even though I set it to be allowed by the firewall it keeps failing with a `Failed to post close command error 1726`... it might be due to some signing issues but I couldn't figure out a way around it except by installing the official package...